Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDeleteAsOverrideTests.cpp

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Prefab/PrefabTestFixture.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+
+namespace UnitTest
+{
+    using PrefabDeleteAsOverrideTests = PrefabTestFixture;
+
+    TEST_F(PrefabDeleteAsOverrideTests, DeleteEntitiesAndAllDescendantsInInstance_DeleteSingleEntitySucceeds)
+    {
+        // Level            <-- focused
+        // | Car_1
+        //   | Tire         <-- delete
+        // | Car_2
+        //   | Tire
+
+        const AZStd::string carPrefabName = "CarPrefab";
+        const AZStd::string tireEntityName = "Tire";
+        const AZStd::string firstCarName = "Car_1";
+        const AZStd::string secondCarName = "Car_2";
+
+        AZ::IO::Path engineRootPath;
+        m_settingsRegistryInterface->Get(engineRootPath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_EngineRootFolder);
+        AZ::IO::Path carPrefabFilepath(engineRootPath);
+        carPrefabFilepath.Append(carPrefabName);
+
+        // Create and rename the first car.
+        AZ::EntityId tireEntityId = CreateEditorEntityUnderRoot(tireEntityName);
+        AZ::EntityId firstCarContainerId = CreateEditorPrefab(carPrefabFilepath, { tireEntityId });
+        EntityAlias tireEntityAlias = FindEntityAliasInInstance(firstCarContainerId, tireEntityName);
+        RenameEntity(firstCarContainerId, firstCarName);
+
+        // Create and rename the second car.
+        AZ::EntityId secondCarContainerId = InstantiateEditorPrefab(carPrefabFilepath, GetRootContainerEntityId());
+        RenameEntity(secondCarContainerId, secondCarName);
+
+        InstanceAlias firstCarInstanceAlias = FindNestedInstanceAliasInInstance(GetRootContainerEntityId(), firstCarName);
+        ASSERT_FALSE(firstCarInstanceAlias.empty());
+
+        InstanceAlias secondCarInstanceAlias = FindNestedInstanceAliasInInstance(GetRootContainerEntityId(), secondCarName);
+        ASSERT_FALSE(secondCarInstanceAlias.empty());
```

Comment:
Do we need this validation of firstCarInstanceAlias and secondCarInstanceAlias? Seems like it should have been caught upon creation if there was an issue. Same question for the other tests.

Reply:
Nice catch! They are not needed.